locality_alignment/backbones/dbot_clip.py

```diff
@@ -0,0 +1,696 @@
+        #         y = r_ids + [0] + dis
+
+        #         t = dst_size // 2.0
+        #         dx = np.arange(-t, t + 0.1, 1.0)
+        #         dy = np.arange(-t, t + 0.1, 1.0)
+
+        #         print("Original positions = %s" % str(x))
+        #         print("Target positions = %s" % str(dx))
+
+        #         all_rel_pos_bias = []
+
+        #         for i in range(num_attn_heads):
+        #             z = rel_pos_bias[:, i].view(src_size, src_size).float().numpy()
+        #             f = interpolate.interp2d(x, y, z, kind="cubic")
+        #             all_rel_pos_bias.append(torch.Tensor(f(dx, dy)).contiguous().view(-1, 1).to(rel_pos_bias.device))
+
+        #         rel_pos_bias = torch.cat(all_rel_pos_bias, dim=-1)
+
+        #         new_rel_pos_bias = torch.cat((rel_pos_bias, extra_tokens), dim=0)
+        #         checkpoint_model[key] = new_rel_pos_bias
+    # pdb.set_trace()
+    # interpolate position embedding
+    if ("pos_embed" in checkpoint_model) and (model.pos_embed is not None):
+        pos_embed_checkpoint = checkpoint_model["pos_embed"]
+        embedding_size = pos_embed_checkpoint.shape[-1]
+        num_patches = model.patch_embed.num_patches
+        num_extra_tokens = model.pos_embed.shape[-2] - num_patches
+        # height (== width) for the checkpoint position embedding
+        orig_size = int((pos_embed_checkpoint.shape[-2] - num_extra_tokens) ** 0.5)
+        # height (== width) for the new position embedding
+        new_size = int(num_patches**0.5)
+        # class_token and dist_token are kept unchanged
+        if orig_size != new_size:
+            print("Position interpolate from %dx%d to %dx%d" % (orig_size, orig_size, new_size, new_size))
+            extra_tokens = pos_embed_checkpoint[:, :num_extra_tokens]
+            # only the position tokens are interpolated
+            pos_tokens = pos_embed_checkpoint[:, num_extra_tokens:]
+            pos_tokens = pos_tokens.reshape(-1, orig_size, orig_size, embedding_size).permute(0, 3, 1, 2)
+            pos_tokens = torch.nn.functional.interpolate(
+                pos_tokens, size=(new_size, new_size), mode="bicubic", align_corners=False
+            )
+            pos_tokens = pos_tokens.permute(0, 2, 3, 1).flatten(1, 2)
+            new_pos_embed = torch.cat((extra_tokens, pos_tokens), dim=1)
+            checkpoint_model["pos_embed"] = new_pos_embed
+
+    custom_load_state_dict(model, checkpoint_model, prefix=model_prefix)
+
+    return model
+
+
+def _create_dbot_vit(variant: str, pretrained: bool = False, **kwargs) -> dBOTCLIPVisionTransformer:
+    kwargs["pretrained_cfg"] = _cfg()
+
+    model = timm.models._builder.build_model_with_cfg(
+        dBOTCLIPVisionTransformer,
+        variant,
+        pretrained=False,
+        **kwargs,
+    )
+
+    if pretrained and variant.startswith("vit_base"):
+        # Load checkpoint.
+        cache_dir = torch.hub.get_dir()
+        file_name = "85.7_clip_base_pre.pth"
```

Comment:
Hardcode filename into strings

Reply:
sorry, what do you mean? Isn't it already hardcoded?

Reply:
saw the comment below, understand what you mean